### ISSUE TITLE
Propose architecture for gamma delays

### ISSUE THREAD

**Issue by kgostic** [epinowcast/epidist#21]
## Goal
Currently the package only estimates lognormal delays. We'd like to add the ability to estimate gamma delays. We'd like to think through how to architect the ability to estimate more than one kind of model.

Specifically, we need functions that can:

* Estimate gamma and lognormal delays
* Process the output of a gamma or lognormal brms fit and transform/retrun the estimated parameters into {mean/sd; distribution-specific parameters on the natural scale; distribution-specific parameters on the log scale}
* Summarize the transformed outputs into quantiles
* Make some plots of the distribution and parameter estimates

**Describe the solution you'd like**
* The easiest approach is to hard-code separate models and post-processing functions for the gamma and lognormal distributions, but this is clunky and inelegant
* Let's read about R classes and brms families to think through whether there's a better architectural solution that can be implemented with relatively light effort while we're starting

**Scope**
* Read about implementation options
* Draft a proposal for implementation in a new issue
* Tag @seabbs and Katie for review before starting to implement

**Comment by seabbs** [comment]
>  {mean/sd; distribution-specific parameters on the natural scal

We discussed whether or not we wanted to do this last week and didn't reach a firm conclusion IMO. I would vote for not providing this as functionality initially until we have a good idea of how to implement it in a generic way (as can easily do it in a given use case with some thought. That being said could easily go either way.

**Comment by kgostic** [comment]
That's fair. I put it here though, because it should be on the long-term horizon guiding our decisions now.

**Comment by seabbs** [comment]
Agree but in the interest of modularisation of issues we agree its out of scope for a PR targeting this issue? 

 I also think the first issue we need here is establishing the framework for just the lognormal and once that is PR'd this can be addressed.
